### PR TITLE
[17506] Partial set subparser

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
This PR includes:

* First set subparser level
* Main participant element parser
* Participant default profile attribute implementation
* External locators parser and implementation for Participant default external locators unicast.
* Test plan with CLI commands and expected outputs in order to test and confirm that fulfills CLI requirements.
* Refactor library include strategy

### CODE REVIEW

File: cli/test/README.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+|`fastddsqosprof file.xml set participant[profile].use_builtin_transports`|**PENDING** (currently `Participant use builtin transports flag configuration not yet supported`)|N/A|
+|`fastddsqosprof file.xml set participant[profile].user_transports`|**PENDING** (currently `Participant user transports configuration not yet supported`)|N/A|
+|`fastddsqosprof file.xml set participant[profile].other`|`ERROR: other element not recognized`|`PARTICIPANT_DEFAULT_PROFILE_USAGE`|
+|`fastddsqosprof file.xml set participant[profile].other -h`|**BUG** (currently `ERROR: other element not recognized` when no error is expected)|`PARTICIPANT_DEFAULT_PROFILE_USAGE`|
+|`fastddsqosprof file.xml set participant[profile].default_profile -h`|N/A|`PARTICIPANT_DEFAULT_PROFILE_USAGE`|
+|`fastddsqosprof file.xml set participant[profile].default_profile --help`|N/A|`PARTICIPANT_DEFAULT_PROFILE_USAGE`|
+|`fastddsqosprof file.xml set participant[profile].default_profile help`|N/A|`PARTICIPANT_DEFAULT_PROFILE_USAGE`|
+|`fastddsqosprof file.xml set participant[profile].default_profile argument1 argument2 help`|N/A|`PARTICIPANT_DEFAULT_PROFILE_USAGE`|
```
Comment (JesusPoderoso) [inline cli/test/README.md:67]: Why are the `argument1` and `argument2` contemplated here?

Is it like it does not matter how many arguments are there, that if the last one is `help` or a help flag, it always prints the last subelement's usage?

File: cli/src/cpp/usages.hpp [hunk 2]
```diff
@@ -19,42 +19,151 @@ namespace eprosima {
...
+
+    Options:
+      help -h --help                CLI participant default profile usage
+
+    The allowed <command> options are:
+      clear                         Erase participant default profile from XML file.
+      print                         Print participant default profile.
+      set                           Set participant profile as default in the XML file.
+
+    The <profile_name> element is MANDATORY and it can be any string (whitespaces are supported if quoted).
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply cli/src/cpp/usages.hpp:105]: Yes, as we have discussed offline, the default profile refactor makes sense that for the clear/print commands, the profile is no longer mandatory. A refactor should be done when print/clear parsers are introduced. `TODO` have been left in the affected code, as well as a footnote in the test plan to be reminded of this issue.

File: cli/src/cpp/usages.hpp [hunk 3]
```diff
@@ -19,42 +19,151 @@ namespace eprosima {
...
+)";
+
+static const char PARTICIPANT_USAGE[] =
+        R"(Participant parser:
+
+    Usage:
+      fastddsqosprof <file> <command> participant[<profile_name>].<subelement> [<values>...]
+      fastddsqosprof <file> <command> participant[<profile_name>].<subelement> (help | -h | --help)
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply cli/src/cpp/usages.hpp:113]: It does not seem to... Adding it :+1:

File: cli/src/cpp/usages.hpp [hunk 4]
```diff
@@ -19,42 +19,151 @@ namespace eprosima {
...
+      use_builtin_transports        Participant use builtin transports flag.
+      user_transports               Participant custom user-defined transports.
+)";
+
+static const char SET_SUBPARSER_USAGE[] =
+        R"(Set subparser: fastddsqosprof <file> set
+
+    Usage:
+      set <element> [<values>...]
+      set (-h | --help)
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply cli/src/cpp/usages.hpp:149]: This usage is used by docopt to implement the second level parsing. Applying your suggestion will prevent docopt to be able to correctly parse the arguments passed.

File: cli/src/cpp/usages.hpp [hunk 5]
```diff
@@ -19,42 +19,151 @@ namespace eprosima {
...
+
+    The allowed <element> types are:
+      datareader                Configure DataReader profile.
+      datawriter                Configure DataWriter profile.
+      help                      Available configuration options for the set command.
+      intraprocess              Configure intra-process delivery.
+      log                       Configure log module.
+      participant               Configure DomainParticipant profile.
+      transport_descriptor      Configure transport descriptor profile.
+      types                     Configure dynamic types.
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply cli/src/cpp/usages.hpp:162]: Good catch!

File: cli/test/README.md (hunk 1 above)
Comment (JLBuenoLopez-eProsima) [reply cli/test/README.md:67]: That is the idea. If someone is using the CLI and have passed some arguments but does not remember what is the next required argument, just by typing help the corresponding usage should be shown.

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Pull request merged**